core/src/main/scala/io/chrisdavenport/rediculous/RedisStream.scala

```diff
@@ -0,0 +1,62 @@
+package io.chrisdavenport.rediculous
+
+import cats.implicits._
+import fs2.{Stream, Pipe}
+import scala.concurrent.duration.Duration
+import RedisCommands.{XAddOpts, XReadOpts, StreamOffset, Trimming, xadd, xread}
+import cats.effect._
+
+final case class XAddMessage(key: String, body: Map[String, String], approxMaxlen: Option[Long] = None)
+final case class XReadMessage(id: MessageId, key: String, body: Map[String, String])
+final case class MessageId(value: String) extends AnyVal
```

Comment:
Considering the other strings in this context, don't think we need this newtype.